dlt/destinations/impl/bigquery/bigquery.py

```diff
@@ -287,12 +288,22 @@ def _get_table_update_sql(
                     " GENERATE_ARRAY(-172800000, 691200000, 86400))"
                 )
 
-        if cluster_list := [
-            self.sql_client.escape_column_name(c["name"])
-            for c in new_columns
-            if c.get("cluster") or c.get(CLUSTER_HINT, False)
-        ]:
+        cluster_columns = cast(Iterable[str], table.get(CLUSTER_COLUMNS_HINT, []))
+        if cluster_columns:
+            cluster_list = [self.sql_client.escape_column_name(col) for col in cluster_columns]
             sql[0] += "\nCLUSTER BY " + ", ".join(cluster_list)
+        else:
+            # Fallback for legacy and test scenarios where only per-column 'cluster' hints are set.
+            # This is required for backward compatibility and to pass tests like
+            # 'test_create_table_with_partition_and_cluster' in test_bigquery_table_builder.py,
+            # which set clustering only via per-column hints and expect the CLUSTER BY clause.
+            cluster_cols_from_columns = [
+                self.sql_client.escape_column_name(c["name"])
+                for c in new_columns
+                if c.get("cluster") or c.get(CLUSTER_HINT, False)
+            ]
+            if cluster_cols_from_columns:
+                sql[0] += "\nCLUSTER BY " + ", ".join(cluster_cols_from_columns)
```

Comment:
@rudolfix 

Do we want to keep this branch to support the old behavior?

My thoughts:
Looking at the `test_create_table_with_partition_and_cluster` test, it's ambiguous to deduce the intended clustering column order if users only set per-column "cluster" hints. The resulting order in the CLUSTER BY clause will just follow the order of columns in the schema, not necessarily the user's intent. If users directly modify the schema or set per-column hints, it's not possible to reliably track or enforce the intended clustering order without introducing more complexity.

Should we encourage users to always use the `bigquery_adapter`'s `cluster` argument for explicit ordering, and discourage direct schema editing for clustering?

Reply:
btw. I bet we'll have similar requests for primary keys, merge keys etc. where we also follow column ordering... maybe I'll add it to our QoL project

Reply:
@rudolfix I've made the deprecation and updated the docs.

Please have a look and let me know if need needs adjusting. I tried to keep it consistent with other deperactions.